### ISSUE TITLE
WritableStream internal assertion error

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#9266]
Comes up in btoa wpt tests as-well.

```
Internal Error: AssertionError: Assertion failed.
    at assert (deno:op_crates/fetch/11_streams.js:25:13)
    at writableStreamFinishInFlightCloseWithError (deno:op_crates/fetch/11_streams.js:2762:5)
    at deno:op_crates/fetch/11_streams.js:2489:7
```

_Originally posted by @caspervonb in https://github.com/denoland/deno/issues/9252#issuecomment-766467996_

**Comment by kitsonk** [comment]
@caspervonb it is a different issue, so breaking it off...

**Comment by kitsonk** [comment]
Hmmm... that is strange, the writable stream shouldn't ever get into that state.  It is being closed, but it is already closed or errored, which it shouldn't be.  I will take a look at the `btoa` test, because nothing else has triggered that particular incorrect state.

**Comment by caspervonb** [comment]
The failure was coming from this fetch https://github.com/web-platform-tests/wpt/blob/581873eb00db0820a0d425dd9c005705cfbbc06f/html/webappapis/atob/base64.any.js#L127-L129

**Comment by caspervonb** [comment]
Was this fixed? no longer occurring in #9053
